generators/Generators.fs

```diff
@@ -871,20 +871,37 @@ type PalindromeProducts() =
         sprintf "(%s, %s)" (string value.[0]) (string value.[1])
 
     let toPalindromeProducts (value: JToken) =
-        let palindromeValue = value.Value<int>("value")
+        let palindromeValue = value.Value("value")
         let factors = 
             value.SelectToken("factors")
             |> List.mapRender toFactors
 
-        sprintf "(%d, %s)" palindromeValue factors
+        match palindromeValue with
+        | null -> sprintf "(None, %s)" factors
+        | _ -> sprintf "(Some %s, %s)" palindromeValue factors
```

Comment:
I think you should be able to use the `Obj.render` function here.

Reply:
I'll look into it, just applying the function to the tuple (palindromeValue, factors) does not produce a good result.

Reply:
Ah, if you don't get it working, then feel free to ignore!

Reply:
I didn't find a clean and easy way to use Obj.render